### PR TITLE
Add private notes feature to newDot

### PR DESCRIPTION
**Pull request by techievivek**
### Details

This PR adds a new feature to newDot, i.e adding privateNotes on chat report.
1. Users should be able to add private notes on each of the chat report.
2. An Expensify and team.expensify.com team member can view private notes of other team.expensify.com members.
3. No other users should be allowed to view other user's note.
### Fixed Issues

$ https://github.com/Expensify/App/issues/25489
PROPOSAL: N/A

### Tests

1. Login to newDot.
2. Go to any DM/ room chat report details page and you will see an option of `private notes` page.
3. Click on the `private notes` page and you will see another option of `my note` on the next page.
4. Once you clicked on that option you should be able to add notes to your chat report.
5. Verify you are able to add notes in Markdown format and it is rendered in HTML.
6. If you are on an expensify.com domain or team.expensify subdomain then you should be able to view private notes of other team.expensify.com members.
- [x] Verify that no errors appear in the JS console

### Offline tests

1. Go offline and click on any chat where you haven't opened the private notes page before, in this case it should show a not found page.
2. Go offline and access the private notes page on a chat report where you have already opened this page, you should be able to see the private note entries and you should be able to update your private note. Once you click on save it should render the content with low opacity.
### QA Steps

Same as tests steps.
- [ ] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/35863227/25de877b-f32a-4c49-86b4-e4d5a7d3aeca

</details>

<details>
<summary>Mobile Web - Chrome</summary>

</details>

<details>
<summary>Mobile Web - Safari</summary>

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/35863227/c4bcd5ce-4344-4f37-82f8-eae3ca5e3874

</details>

<details>
<summary>iOS</summary>

</details>

<details>
<summary>Android</summary>

</details>

### CODE REVIEW

**Comment by techievivek** [comment]
Adding design label to get some review from Shawn for these 2 points.
1. Since we have set the maxHeight we see a scrollbar appearing after adding 3-4 lines of text, maybe we can increase the maxHeight here?
Screenshot 2023-08-30 at 1 59 35 AM

2. Maybe we can add a border at the bottom in the view mode? For now it doesn't look like it can be edited.


https://github.com/Expensify/App/assets/35863227/b7214f3d-2431-4137-9e14-ea852bc8a3c7

**Comment by neil-marcellini** [comment]
I'll review after C+ approves. Please DM me when that happens.

**Comment by shawnborton** [comment]
> 2. Maybe we can add a border at the bottom in the view mode? For now it doesn't look like it can be edited.

I thought we decided on Slack that there is no need for the extra tap to be able to edit? Can you dig up those conversations? 

That being said, I would think you would just display the textarea as soon as you navigated there.

**Comment by shawnborton** [comment]
Ah got it, but it looks like my comment was missed. We should use the push-to-page pattern to get to the edit state, not just changing the component in-place. Let me know if that makes sense.

**Comment by techievivek** [comment]
Ok, just found something here: https://github.com/Expensify/App/issues/25499, going to take a look to figure out what needs to be done to have that behaviour.

**Comment by shawnborton** [comment]
Yup! We use it everywhere, for example, on the profile page:

Profile Page | Push-to-Page
------ | ------
image | image

**Comment by techievivek** [comment]
@shawnborton We already use the push-to-page pattern in the listing page and when the user clicks on the note we take them to view/edit the note, are you suggesting we use an extra push-to-page iteration to take them to edit mode?


https://github.com/Expensify/App/assets/35863227/feb4bce6-eb10-46db-abaf-e555e94a0eaf

**Comment by shawnborton** [comment]
Yup, that's correct.

**Comment by techievivek** [comment]
@shawnborton Interesting! Given this scenario, what should be the ideal number of content lines to display on the intermediary page? Also, how would a user access the complete note in the HTML-rich content view? Let me know if you want me to bring this on slack to resolve it quickly.

**Comment by techievivek** [comment]
CC @garrettmknight, as this is different from what we discussed originally. In our original design we decided to render a single page where user can switch between view and edit mode.

**Comment by shawnborton** [comment]
> Given this scenario, what should be the ideal number of content lines to display on the intermediary page?

I personally prefer not to truncate any information when in read-only mode, so it would just be the push-to-page row but it would display all of the notes that it can. This is similar to how we handle the description field for tasks, give it a shot! Then once you tap into the push row to edit the text area, we might have a standard textarea that is capped at a certain height or something.

**Comment by techievivek** [comment]
> This is similar to how we handle the description field for tasks, give it a shot! 

Perfect, this looks great. I will update to follow the same pattern here, thanks. 

Screenshot 2023-08-31 at 2 23 13 PM

**Comment by dannymcclain** [comment]
I might be missing something, but I'm not seeing the push-to-page interaction for the note.

Also, not sure what all markdown we're planning to support, but I'm seeing some things work (Heading 1, Link) and some things not (Heading 2-5, bulleted lists). Is this intentional for this rollout?

A couple other things I noticed:
- the main notes screen that displays my note does not scroll (for me) so I cant see my whole note
- There's no way to clear/delete/remove the notes I have—if I go into edit mode and delete all my note text, the save button doesn't do anything

@shawnborton would you mind also checking this out? (sorry to be a bother!)

image

**Comment by shawnborton** [comment]
Oh and good catch on the save button not working properly too! @techievivek can you take a look at that one too?

**Comment by techievivek** [comment]
This is ready for review @sobitneupane @neil-marcellini @dannymcclain. I will update the GH with updated screenshots.

**Comment by dannymcclain** [comment]
This is looking/working much better. Looks like you addressed all the issue I brought up in my previous comment 🙌

My only little piece of feedback is that our push-to-page component usually goes edge to edge and does not have any left/right padding. You can see the difference in the attached screenshots.

Screenshot 2023-09-12 at 10 20 09 AM

Screenshot 2023-09-12 at 10 19 57 AM

**Comment by techievivek** [comment]
@dannymcclain 

> My only little piece of feedback is that our push-to-page component usually goes edge to edge and does not have any left/right padding.

:100: agree, I have updated the code.

**Review by dannymcclain — DISMISSED** [review]
I think this looks good to me—just want to note that at some point we may want to increase the height of the actual text area that you type notes into. But honestly I don't think it's a big deal right this second. I'd kinda wanna wait and see how people use this before customizing components and stuff.

**Comment by neil-marcellini** [comment]
I'll review after C+ approves

**Review by sobitneupane — COMMENTED** [review]
@techievivek 

We had discussed about hiding private notes in thread and task. But they are still present. Are we going to keep them?
Screenshot 2023-09-13 at 16 25 58

Screenshot 2023-09-13 at 16 25 24

**Comment by techievivek** [comment]
@sobitneupane Just added the condition to not show private notes on thread and tasks.

**Review by sobitneupane — COMMENTED** [review]
@techievivek 

When I open Private Note for the first time. The flow is:
Private Notes > My note > Note Input (If there is no note)
Private Notes > My note > Note Preview > Note Input (If there is a note)

But on second run (may be after adding a private note and removing it). The flow is:
Private Notes > Note Input (If there is no note)
Private Notes > Note Preview > Note Input (If there is a note)

Basically Page with My note option doesn't appear consistently.

https://github.com/Expensify/App/assets/25876548/662171fc-562f-4bf3-874e-70ddd3b34eb1

2. Even Note Preview Page miss out sometime. And the flow becomes Private Notes > Note Input (even if there is a note). If we close the RHP and reopen the note page, it gets fixed the flow becomes Private Notes> Note Preview > Note Input

https://github.com/Expensify/App/assets/25876548/cfe2d752-6402-43c2-bc00-164da281dbb4

3. And the other issue is:
After Note is added and removed by saving empty note, we show preview for empty note which gets removed if we close and remove RHP.

https://github.com/Expensify/App/assets/25876548/6768a093-8fd1-424d-b592-ca184ba12b4b

**Comment by techievivek** [comment]
@sobitneupane 

<blockquote>
When I open Private Note for the first time. The flow is:
Private Notes > My note > Note Input (If there is no note)
Private Notes > My note > Note Preview > Note Input (If there is a note)

But on second run (may be after adding a private note and removing it). The flow is:
Private Notes > Note Input (If there is no note)
Private Notes > Note Preview > Note Input (If there is a note)
</blockquote>

This was on purpose. But I agree that behaviour gets inconsistent sometimes because of the implementation of `<` button. 

`My note` part gets shown only when there are multiple notes that you have access to, when you open the note for the first time since we don't have anything in Onyx, we make a network call and show the private notes list on the listing page. And that's the reason why that page comes into view in the first attempt. In other cases, since there is just a single note we take the user right away to privateNotesViewPage or the edit page(based on whether or not the content is empty). But I think we can just show the `My note` page even if it's a single note to keep things consistent. So I will make that improvement.


> Even Note Preview Page miss out sometime. And the flow becomes Private Notes > Note Input (even if there is a note)

Hmm, this is new, I would expect it to show the view page if the content is not empty. Let me test/fix this quickly.

> And the other issue is:
After Note is added and removed by saving empty note, we show preview for empty note which gets removed if we close and remove RHP.

Ok , I see this issue, I will need to navigate back to the view page only if the content is not empty and not every time when we press the `<` button.

**Comment by techievivek** [comment]
I invested some time and tried making the navigation consistent, but since I am conditionally navigating(only when the content is non-empty) to PrivateNotesViewPage this is making things behave weirdly. @dannymcclain Do you have any suggestion on what we can show on the privateNotesViewPage if the private notes content is empty?

This is how it looks like when the private note content is empty for the user.
Screenshot 2023-09-14 at 6 19 51 PM

**Comment by dannymcclain** [comment]
Would love to get feedback on this from @shawnborton — I don't hate the idea of having a little graphic in there, but I'm wondering if it's a bit _too_ heavy for what this is.

**Comment by shawnborton** [comment]
Yeah I agree, given that this is a very functional power user feature, I think we can go without a fancy empty state for now. So I would vote to just keep it as is.

**Comment by techievivek** [comment]
@shawnborton What do you suggest we do if the private note content is empty? 

https://user-images.githubusercontent.com/35863227/267979586-c4ebe563-75c0-4a81-a51d-9bceaf8a6695.png

I can render them as it is, right?

**Comment by shawnborton** [comment]
What does the screen before that look like? Can you show me a series of sceens in a flow to get here?

**Comment by techievivek** [comment]
Sharing a screen recording, thanks.

**Comment by techievivek** [comment]
@shawnborton This is what the flow looks like.
https://github.com/Expensify/App/assets/35863227/eb9a2226-7e8e-49e2-9aef-aeb0fe98c4cb

**Comment by shawnborton** [comment]
Cool, yeah I think that's fine as it is.

**Comment by techievivek** [comment]
Ok, awesome. I was trying to over-optimize it. 😄 :ohnothing:

**Comment by techievivek** [comment]
@sobitneupane Made the navigation behaviour consistent. Also, I was wondering if you could please take a look at the whole PR in one go. This would be a big help in keeping things moving smoothly and avoiding unnecessary delays. It can get a bit tricky when we're making changes one by one and constantly going back and forth with reviews. Since we're looking forward to getting this shipped soon, it would be awesome if you could give the entire PR a once-over and share your feedback/comments/suggestions all at once. Thanks

**Review by sobitneupane — COMMENTED** [review]
@techievivek Overall Changes looks good. Are following expected?

1. We don't autofocus the textinput for Private Note.
2. If offline, Not Found Page is shown on clicking Private Note.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/25876548/57eac1c9-fe47-44c4-9df5-3d489a04c835

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/25876548/45cf0091-f98a-4ecb-aaa9-27634888646f

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/25876548/50cb0452-396c-49fc-bc2b-f95daa890eee

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/25876548/7fe28ce6-eece-466b-8d58-7f82c550edc5

</details>

<details>
<summary>iOS</summary>



https://github.com/Expensify/App/assets/25876548/931197e4-c699-4851-bff3-8ccf464d9438

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/25876548/88acf8cb-6272-47f3-be9e-29853a2c49c9

</details>

**Comment by techievivek** [comment]
> We don't autofocus the textinput for Private Note.

We should/can definitely do this.

> If offline, Not Found Page is shown on clicking Private Note.

If notes are not present in the Onyx, then this is expected(because we are not sure if the user has access to the report or not or if they have deep-linked to a random report's private note) otherwise it is a bug. Let me check the videos to confirm that.

**Comment by techievivek** [comment]
@sobitneupane 

> If notes are not present in the Onyx, then this is expected(because we are not sure if the user has access to the report or not or if they have deep-linked to a random report's private note) otherwise it is a bug. Let me check the videos to confirm that.

Can you confirm if this was the case? In meanwhile I will add the auto-focus on edit page.

**Comment by sobitneupane** [comment]
> If notes are not present in the Onyx, then this is expected(because we are not sure if the user has access to the report or not or if they have deep-linked to a random report's private note) otherwise it is a bug. Let me check the videos to confirm that.

Yes. This is the case only when accessing Private Note of a report for the first time. We are good.

**Comment by techievivek** [comment]
Nice, then we are good, thanks for confirming.

**Comment by techievivek** [comment]
Auto-focus in action.


https://github.com/Expensify/App/assets/35863227/826056cf-439e-455e-898d-2713d44dcf33

**Comment by sobitneupane** [comment]
@techievivek `textinput` doesn't scroll to the last line if there are more than 4 lines in input.


https://github.com/Expensify/App/assets/25876548/bcb1746c-0c3b-4ced-9f0d-21d11c40e5ae

**Comment by techievivek** [comment]
@sobitneupane Did you mean auto-scrolling to bring the last line in focus?

**Comment by sobitneupane** [comment]
@techievivek Yes.

**Comment by techievivek** [comment]
Do you know if we have a similar pattern in our App that I can look at to fix this?

**Comment by techievivek** [comment]
Otherwise, we can ship this MVP and I can work on these improvements separately because I don't think it should be a blocker here?

**Comment by sobitneupane** [comment]
@techievivek The description text in Task page autoscrolls to bring the last line in focus.

**Comment by techievivek** [comment]
Nice, that works like a charm, let me take a look at the code. Thanks for the pointer 🙌

**Comment by techievivek** [comment]
Wow, someone has already created an utility function for this `focusAndUpdateMultilineInputRange`, great. Adding it in action.

**Comment by techievivek** [comment]
Ok, just tested auto-focus and auto-scroll and it seems to work well now.

https://github.com/Expensify/App/assets/35863227/9df24407-6a17-46b9-9ba5-840dc5aa054f

**Review by sobitneupane — APPROVED** [review]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/25876548/7e4c8a04-b670-4f16-a00e-1387d17dc479

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/25876548/5d78e84f-e9fc-4104-bcce-ad8cb795a85d

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/25876548/00f17f02-9c8b-401b-b6d1-41faf05178cf

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/25876548/4776f729-5660-4f4c-b871-b5fb7e099c90

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/25876548/9dbf2369-c84e-4203-a8d0-097c9fd37fdc

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/25876548/3b4963c8-adb7-45b2-8c23-c057176a658f

</details>

**Comment by techievivek** [comment]
Neil seems to be ooo, so I have reached out to Jules to give it a review.

**Review by Julesssss — APPROVED** [review]
Approving as it tests well, but I find the double menu confusing and redundant TBH. Going to share that in Slack

**Comment by techievivek** [comment]
>  I find the double menu confusing and redundant TBH

Yeah, I agree this seems an extra step there for no reason.
I do have an alternate there but I think that doesn't match our design system https://github.com/Expensify/App/pull/25761#issuecomment-1699140676

File: src/pages/PrivateNotes/PrivateNotesListPage.js
Comment (eVoloshchak) [inline src/pages/PrivateNotes/PrivateNotesListPage.js:134]: Coming from https://github.com/Expensify/App/issues/27590
The notes list didn't take up the full screen height, which resulted in offline indicator not being positioned at the bottom
We resolved this by adding a ScrollView (in case there are more notes that fit on the screen) with `flexGrow: 1` style applied

**Pull request merged**
